services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -254,7 +254,7 @@ public void authenticationFinished(AuthenticationSessionModel authSession, Broke
 
     @Override
     public Response retrieveToken(KeycloakSession session, FederatedIdentityModel identity) {
-        return Response.ok(identity.getToken()).build();
+        return Response.ok(identity.getToken()).type(MediaType.TEXT_PLAIN_TYPE).build();
```

Comment:
The response is basically the raw SAML Response (encoded).

Reply:
We should always set a media type. It is a common best practice.

We did not catch that before because Resteasy was setting `application/octet-stream` which is kinda worse ...

Making sure we set `text/plain` is at least more aligned with the type of the response.